File: cli-tools/automation/python/get_job_status.py
import argparse
import requests
import json
import sys
from utils import BASE_URL, get_access_token

def get_status(job_id):
    try:
        token = get_access_token()
        headers = {
            'Authorization': f'Bearer {token}'
        }
        
        response = requests.get(f"{BASE_URL}/images/jobs/{job_id}/status", headers=headers)
        response.raise_for_status()
        
        print(json.dumps(response.json(), indent=2))
        
    except Exception as e:
        print(f"Error: {e}", file=sys.stderr)

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='Get Job Status from Pixelz Automation API')
    parser.add_argument('--job', required=True, help='Job ID')
    args = parser.parse_args()
    
    get_status(args.job)

File: cli-tools/automation/node/get-job-status.js
const axios = require('axios');
const { BASE_URL, getAccessToken } = require('./utils');

const args = process.argv.slice(2);
const jobId = args.find(arg => arg.startsWith('--job='))?.split('=')[1];

if (!jobId) {
    console.log("Usage: node get-job-status.js --job=<job_id>");
    process.exit(1);
}

async function getStatus() {
    try {
        const token = await getAccessToken();
        const response = await axios.get(`${BASE_URL}/images/jobs/${jobId}/status`, {
            headers: { 'Authorization': `Bearer ${token}` }
        });

        console.log(JSON.stringify(response.data, null, 2));

    } catch (error) {
        console.error("Error:", error.message);
        if (error.response) console.error(error.response.data);
    }
}

getStatus();
